src/libcore/borrow.rs

```diff
@@ -12,26 +12,143 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
-/// A trait for borrowing data.
+// impl Borrow<str> for String
+// impl<T> Borrow<T> for Arc<T>
+// impl<K> HashSet<K> { fn get<Q>(&self, q: &Q) where K: Borrow<Q> }
+
+/// A trait identifying how borrowed data behaves.
+///
+/// If a type implements this trait, it signals that a reference to it behaves
+/// exactly like a reference to `Borrowed`. As a consequence, if a trait is
+/// implemented both by `Self` and `Borrowed`, all trait methods that
+/// take a `&self` argument must produce the same result in both
+/// implementations.
+///
+/// As a consequence, this trait should only be implemented for types managing
+/// a value of another type without modifying its behavior. Examples are
+/// smart pointers such as [`Box`] or [`Rc`] as well the owned version of
```

Comment:
these three types should have their `<T>`s

Reply:
Ack.

Before I commit that: `AsRef` and `Borrow` don’t need theirs?

Reply:
i always forget that those don't have parameters, but generally, they're part of the type's "proper name", so they get paramters. So like "vector" doesn't need one, but `Vec<T>` should have one, that kind of thing